features/2025-07-10-offline-install-feature-spec.md

```diff
@@ -0,0 +1,407 @@
+  name: 'tf'
+  terraformCLI: {
+    url: 'https://<TF_MIRROR_URL>/terraform_1.5.7_linux_amd64.zip'
+    checksum: 'sha256:37f2d497ea512324d59b50ebf1b58a6fcc2a2828d638a4f6fdb1f41af00140f3'
+  }
+  terraformrc: {
+    provider_installation: {
+      network_mirror: {
+        path: 'https://<TF_MIRROR_URL>/'
+        include: ["*"]
+      }
+      direct: {
+        exclude: ["azurerm"]
+      }
+    }
+    // Equivilent to recipeConfig.authentication on Environments today
+    credentials: {
+      <TF_MIRROR_URL>: {
+        secret: providerSecret.id
+      }
+    }
+    // Environment variables injected into the Terraform process
+    // Equivilent to recipeConfig.env on Environments today
+    env: {
+      TF_REGISTRY_CLIENT_TIMEOUT: 15
+    }
+  }
+}
+
+resource providerSecret 'Applications.Core/secretStores@2023-10-01-preview' = {
+  name: 'providerSecret'
+  properties: {
+    type: 'generic'
+    data: {
+      token: {
+        value: token
+      }
+    }
+  }
+}
+```
+
+The Terraform resource intentionally does not include these terraformrc properties:
+
+* `credentials_helper`
+* `provider_installation.filesystem_mirror`
+* `disable_checkpoint`
+* `disable_checkpoint_signature`
+* `plugin_cache_dir`
+
+The platform engineer deploys the Terraform resource.
+
+````bash
+$ rad deploy terraform.bicep
+````
+
+When the Terraform resource is deployed, Radius:
+
+* Downloads the specified Terraform binary and validates it against the provided checksum
+* Installs the Terraform binary in the appropriate location within the Radius control plane
+* Performs a `terraform init` and reports any output to the platform engineer
+
+> [!NOTE]
+>
+> This feature specification diffs from the initial implementation which took properties on the `rad install` command. 
+
+The Environment resource no longer has a recipeConfig. This includes:
+
+| As-Is Resource | As-Is Property                          | To-Be Resource | To-Be Property            |
+| -------------- | --------------------------------------- | -------------- | ------------------------- |
+| Environment    | `recipeConfig.terraform.authentication` | Terraform      | `terraformrc.credentials` |
+| Environment    | `recipeConfig.terraform.providers`      | Terraform      | TODO                      |
+
+> [!CAUTION]
+>
+> **TODO**: Add details on how to handle `recipeConfig.terraform.providers`. This property is used to pass credentials to other providers such as AzureDevOps. Radius only passes credentials to Azure, AWS, and Kubernetes Terraform providers.
+
+#### Alternatives Considered
+
+We considered extending the `recipeConfig` property of the Environment resource to include the provider mirror. The challenge with this approach is that the `recipeConfig` is on the Environment. As with Recipe Packs, the Environment resource needs to be as slim as possible because there will be hundreds of environments but only one configuration for Terraform.
```

Comment:
Another option would be to encapsulate the `recipeConfig` into a [`.bicepparam`](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameter-files?tabs=Bicep) file that can be reused across the Environments.